stroyprombeton/context.py

```diff
@@ -131,43 +65,54 @@ def category(self):
             id=self.request_data.id
         )
 
-    # @todo #431:15m  Create type hints for context on refarm side. se2
-    def context(self) -> dict:
-        all_tags = newcontext.Tags(models.Tag.objects.all())
+    @property
+    def tags(self) -> newcontext.Tags:
+        return newcontext.Tags(models.Tag.objects.all())
+
+    def select_tags(self) -> newcontext.Tags:
+        all_tags = self.tags
 
         selected_tags = newcontext.tags.ParsedTags(
             tags=all_tags,
             raw_tags=self.request_data.tags,
         )
         if self.request_data.tags:
             selected_tags = newcontext.tags.Checked404Tags(selected_tags)
+        return selected_tags
 
-        products = (
+    def filter_products(self) -> ProductQuerySet:
+        return (
             models.Product.objects.active()
             .bind_fields()
             .filter_descendants(self.category)
-            .tagged_or_all(selected_tags.qs())
+            .tagged_or_all(self.select_tags().qs())
             .order_by(*settings.PRODUCTS_ORDERING)
         )
 
+    def slice_products(self) -> newcontext.products.PaginatedProducts:
         """
-        We have to use separated variable for pagination.
+        We have to use separated variable/method for pagination.
 
         Because paginated QuerySet can not used as QuerySet.
         It's not the most strong place of Django ORM, of course.
         :return: ProductsContext with paginated QuerySet inside
         """
-        paginated = newcontext.products.PaginatedProducts(
-            products=products,
+        return newcontext.products.PaginatedProducts(
+            products=self.filter_products(),
             url=self.request_data.request.path,
             page_number=self.request_data.pagination_page_number,
             per_page=self.request_data.pagination_per_page,
         )
 
-        images = newcontext.products.ProductImages(paginated.products, Image.objects.all())
-        brands = newcontext.products.ProductBrands(paginated.products, all_tags)
+    # @todo #443:15m  Create Context type on refarm side. se2
```

Comment:
This is little bit unclear.  Where will we use it? How will the type differ from `Context` class?

Reply:
@artemiy312 , i mean ContextDict type. Fixed task